File: module/RnApplication/src/main/java/com/pds/application/BaseApplication.java
package com.pds.application;

import android.content.Context;

import androidx.multidex.MultiDexApplication;

import com.facebook.react.ReactApplication;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.shell.MainReactPackage;

import java.lang.reflect.InvocationHandler;
import java.util.Arrays;
import java.util.List;

/**
 * @author: pengdaosong
 * @CreateTime: 2020/9/26 2:07 PM
 * @Email: pengdaosong@medlinker.com
 * @Description: 在这里初始化相关sdk
 */
public class BaseApplication extends MultiDexApplication implements ReactApplication{
    private static BaseApplication sApplication;
    private InvocationHandler mHandler;

    public static BaseApplication getApplication() {
        return sApplication;
    }

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
    }

    @Override
    public void onCreate() {
        super.onCreate();
        sApplication = this;
    }

    private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
        @Override
        public boolean getUseDeveloperSupport() {
            return BuildConfig.DEBUG;
        }

        @Override
        protected List<ReactPackage> getPackages() {
            return Arrays.<ReactPackage>asList(
                    new MainReactPackage()
            );
        }

        @Override
        protected String getJSMainModuleName() {
            return "index";
        }
    };

    @Override
    public ReactNativeHost getReactNativeHost() {
        return mReactNativeHost;
    }
}

File: IM/socket_app/src/main/java/blog/pds/com/socket/app/common/base/BaseApplication.kt
package blog.pds.com.socket.app.common.base

import androidx.multidex.MultiDexApplication
import blog.pds.com.socket.control.config.ApiManager

/**
 * @author: pengdaosong
 * CreateTime:  2019-06-22 17:57
 * Email：pengdaosong@medlinker.com
 * Description:
 */
class BaseApplication : MultiDexApplication(){

    companion object{
        private lateinit var application: BaseApplication

        const val ip = "192.168.2.2"
        const val port = 6666

        fun app(): BaseApplication {
            return application
        }
    }

    override fun onCreate() {
        super.onCreate()
        application = this
        ApiManager.ipAndPort(
            ip,
            port
        )
    }

}